### DISCUSSION TITLE
Does the Gitbook supports setting the website's light and dark theme according to the system settings?

### DISCUSSION THREAD

**Discussion by mendax1234** [GitbookIO/community#227] [2. Published content]
### What's your feature/idea?

l think Gitbook should change its theme according to the system setting, or according to the time of sunrise and sunset. l don't know whether Gitbook now supports this. Since l find that the theme can only be toggled by the visitor himself.

### What's your desired outcome?

Gitbook can change its theme automatically.

### What's the impact of this for you?

- [ ] Critical
- [X] High
- [ ] Medium
- [ ] Low

**Comment by mendax1234** [comment]
Is there any plan for it? l think this discussion might be forgot 😂

**Reply by aikoii** [reply]
Hi @mendax1234!

Thank you for sharing your thoughts! When it comes to light and dark modes, it's important to remember that everyone has their own preferences. Some people may prefer one mode over the other all the time. That's why we offer the option for readers to choose their preferred mode manually. This helps us accommodate different needs and preferences, making our platform accessible to more people! 😊

Alternatively, you can hide toggle mode from your published docs and set it to default.

**Reply by vfonic** [reply]
I wonder how do you know that _not supporting_ OS system color setting makes your platform "accessible to more people"?

Just like OP, I prefer light mode during the day and dark mode during the night. That's what works on most of the sites I visit.

Some sites have an option that, if user clicks on the dark/light switch, his preference gets remembered. Or it's a account preferences option.

Sites that support OS dark/light mode:
1. Notion
2. Google.com
3. [Shopify dev docs](https://shopify.dev/docs/api/admin-graphql)
4. Trello
5. YouTube
6. GitHub
7. Stack Overflow
8. [Bullet Train docs](https://bullettrain.co/docs)

These are just some from the top of my head.

You can have both functionalities working together so I don't see why this wouldn't be considered?
Why do all these other sites have this functionality? Perhaps they are not all wrong for adding support for that?

**Reply by aikoii** [reply]
Thanks for sharing your thoughts, @vfonic! Apologies for the confusion earlier 😬. Although it's a possibility for the future, at this time, it's not our main priority. I'm leaving it open for now 👀.

**Reply by vfonic** [reply]
Thanks sir! Would you be interested in PR if I find the time to start digging in?

**Reply by johnnymatthews** [reply]
This should be trivial for GitBook to implement using [`prefers-color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme). For example:

```
body {
  background-color: white;
  color: black;
}

@media (prefers-color-scheme: dark) {
  body {
    background-color: black;
    color: white;
  }
}

```

**Reply by addisonschultz** [reply]
Thanks @johnnymatthews @vfonic - we appreciate your thoughts! We'd love to try and prioritise this for a coming cycle of ours, but I can't promise when we'll be able to get to it. It isn't possible to submit PRs for this part of our product (although, feel free to check out our [open source integration platform](https://github.com/GitbookIO/integrations) 😄), but as @aikoii mentioned we'll keep this ticket open until we can get to it!

**Comment by mendax1234** [comment]
@aikoii So, does gitbook support this function now?

**Reply by johnnymatthews** [reply]
From what I can tell GitBook still defaults to a light colour-scheme, until the user changes the toggle. I'm assuming this setting is being set using the [Web Storage API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API).

So to answer your question, no GitBook does not support this feature yet.

**Reply by addisonschultz** [reply]
Unfortunately at this time OS colour scheme isn't supported 😕
